contracts/core/Core.sol

```diff
@@ -59,6 +65,15 @@ contract Core is
         _;
     }
 
+    // Checks to make sure a valid token address is given.
+    modifier isValidTokenAddress(address _tokenAddress) {
+        // Confirm address is not null
+        require(_tokenAddress != address(0));
+        // Confirm address is not Core address
+        require(_tokenAddress != address(this));
```

Comment:
Core will either need to hold a mapping of valid set tokens or query factory for it. Instead of checking that the address is not `this`, it should just check that mapping.

Reply:
seems like it can be handled by @felix2feng 's change